### ISSUE TITLE
关于论文结果的疑问

### ISSUE THREAD

**Issue by xing-wei-zeng** [taiyu-zhu/StepFinder#3]
> 您好！看了您的工作觉得很有意思，想接着做一下这个方向。我尝试按照 Readme 说明进行复现，但是没得到论文的结果，想咨询一下是否我哪里设置有问题？以下是我的运行环境/配置：
> GPU： NVIDIA-SMI 580.76.05              Driver Version: 580.76.05      CUDA Version: 13.0    NVIDIA vGPU-32GB
> 依赖包：
>   torch：2.5.1+cu124
>   transformers：4.53.3
>   numpy：2.1.3
> 对于 Algorithm 数据集使用默认的设置：
>     # Model
>     parser.add_argument("--content_dim", type=int, default=128)
>     parser.add_argument("--agent_dim", type=int, default=32)
>     parser.add_argument("--hidden_dim", type=int, default=64)
>     parser.add_argument("--num_heads", type=int, default=2)
>     parser.add_argument("--head_dim", type=int, default=32)
>     parser.add_argument("--alpha", type=float, default=0.1)
>     parser.add_argument("--beta", type=float, default=0.9)
>     parser.add_argument("--gamma", type=float, default=0.4)
>     parser.add_argument("--scales", type=int, nargs="+", default=[1, 2])
>     parser.add_argument("--dropout", type=float, default=0.5)
>     # Training
>     parser.add_argument("--seed", type=int, default=42)
>     parser.add_argument("--epochs", type=int, default=50)
>     parser.add_argument("--batch_size", type=int, default=16)
>     parser.add_argument("--lr", type=float, default=1e-3)
>     parser.add_argument("--weight_decay", type=float, default=1e-5)
>     parser.add_argument("--lambda_temporal", type=float, default=0.9)
>     parser.add_argument("--patience", type=int, default=10)
> 
> 运行命令：python main.py     --train_dir ./data/Algorithm-Generated/train     --test_dir ./data/Algorithm-Generated/test     --save_path model/Algorithm-Generated_best_model.pth
> 
> <img width="2674" height="1672" alt="Image" src="https://github.com/user-attachments/assets/1ad2f20b-d935-4407-8922-751492844217" />
> 
> 感谢🙏 

 _Originally posted by @xing-wei-zeng in [#2](https://github.com/taiyu-zhu/StepFinder/issues/2#issuecomment-5646433781)_

**Comment by taiyu-zhu** [comment]
您好，我们重新核对并复现了实验。README 中给出了依赖的最低版本要求，我们实际验证使用的具体环境如下：

运行环境
- 操作系统：Ubuntu 22.04.3 LTS
- GPU：NVIDIA GeForce RTX 3080 Ti，12 GB
- NVIDIA Driver：595.71.05
- Python：3.12.14
- PyTorch：2.3.0+cu121
- PyTorch CUDA：12.1
- cuDNN：8.9.2
- Transformers：4.51.3
- NumPy：1.26.4

Embedding 模型
- 模型：Qwen/Qwen3-Embedding-0.6B
- Revision：97b0c614be4d77ee51c0cef4e5f07c00f9eb65b3
- model.safetensors SHA256：0437e45c94563b09e13cb7a64478fc406947a93cb34a7e05870fc8dcd48e23fd

建议固定上述模型 revision，而不是只指定 Qwen3-Embedding-0.6B。模型版本、PyTorch/Transformers 版本及 CUDA 计算环境的差异，都可能导致生成的特征缓存存在数值差异。
如果按照上述环境和模型版本仍然存在问题，欢迎继续在本 Issue 中反馈，我们会进一步协助排查。

**Comment by xing-wei-zeng** [comment]
您好！感谢您的回复，我跟你回复重新租用了服务器进行了全新的实验，以下是实验的配置：

<img width="1728" height="1470" alt="Image" src="https://github.com/user-attachments/assets/e270ef2d-8e88-40d4-b000-40d143b7a956" />

<img width="1738" height="796" alt="Image" src="https://github.com/user-attachments/assets/4ebec698-1a26-4d48-a254-ea69b099aa33" />

<img width="1774" height="356" alt="Image" src="https://github.com/user-attachments/assets/87a29d32-e0c0-4a62-8495-2ae953d8f7f7" />

<img width="2324" height="1166" alt="Image" src="https://github.com/user-attachments/assets/e9527870-a428-4095-a072-6f1bdf46ab1f" />

以下是运行结果：

<img width="2564" height="1172" alt="Image" src="https://github.com/user-attachments/assets/e17cdb40-7e86-476d-a715-f1a0f5eab978" />

<img width="2236" height="1288" alt="Image" src="https://github.com/user-attachments/assets/5437a096-574b-4d06-b483-d5f04458657d" />

不知道是我数据集配置有问题么？我是下载的原版 who&when，以下是数据截图：

<img width="3260" height="1290" alt="Image" src="https://github.com/user-attachments/assets/5e36e2de-5b88-4d6f-821f-ac90e25c70b8" />

再次感谢您的回复～

**Comment by taiyu-zhu** [comment]
您好，非常感谢您的反馈。我们检查后发现，您的数据和环境没有问题。之前的代码只读取每个步骤中的 is_mistake，但原始 Who&When 测试集使用的是 mistake_step，因此在使用 Qwen-Embedding 对测试数据进行特征编码时，生成的缓存文件中标签不正确。现在代码已经修复，请拉取最新代码，删除之前由 Qwen-Embedding 编码生成的测试集特征缓存，然后重新进行特征编码和测试。如果还有问题，欢迎继续反馈。

**Comment by xing-wei-zeng** [comment]
感谢您的回复，我已复现论文的结果！
